File: sample-arcore/app/src/main/java/com/seerslab/argear/sample/ui/FilterFragment.java
package com.seerslab.argear.sample.ui;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.seerslab.argear.sample.model.CategoryModel;
import com.seerslab.argear.sample.viewmodel.ContentsViewModel;
import com.seerslab.argear.sample.R;
import com.seerslab.argear.sample.api.ContentsResponse;
import com.seerslab.argear.sample.model.ItemModel;
import com.seerslab.argear.sample.ui.adapter.FilterListAdapter;

import java.util.List;

public class FilterFragment
        extends Fragment
        implements View.OnClickListener, FilterListAdapter.Listener {

    private static final String TAG = FilterFragment.class.getSimpleName();

    private FilterListAdapter mFilterListAdapter;
    private ContentsViewModel mContentsViewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_filter, container, false);

        rootView.findViewById(R.id.close_filter_button).setOnClickListener(this);
        rootView.findViewById(R.id.clear_filter_button).setOnClickListener(this);
        rootView.findViewById(R.id.filter_plus_button).setOnClickListener(this);
        rootView.findViewById(R.id.filter_minus_button).setOnClickListener(this);
        rootView.findViewById(R.id.vignett_button).setOnClickListener(this);
        rootView.findViewById(R.id.blur_button).setOnClickListener(this);

        RecyclerView recyclerViewFilter = rootView.findViewById(R.id.filter_recyclerview);

        recyclerViewFilter.setHasFixedSize(true);
        LinearLayoutManager filterLayoutManager = new LinearLayoutManager(getContext());
        filterLayoutManager.setOrientation(LinearLayoutManager.HORIZONTAL);
        recyclerViewFilter.setLayoutManager(filterLayoutManager);

        mFilterListAdapter = new FilterListAdapter(getContext(), this);
        recyclerViewFilter.setAdapter(mFilterListAdapter);

        return rootView;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        if (getActivity() != null) {
            mContentsViewModel = new ViewModelProvider(getActivity()).get(ContentsViewModel.class);
            mContentsViewModel.getContents().observe(getViewLifecycleOwner(), new Observer<ContentsResponse>() {
                @Override
                public void onChanged(ContentsResponse contentsResponse) {

                    if (contentsResponse == null) return;

                    for (CategoryModel model : contentsResponse.categories) {
                        if (TextUtils.equals(model.title, "filters")) {
                            mFilterListAdapter.setData(model.items);
                            return;
                        }
                    }

                }
            });
        }
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.close_filter_button:
                getActivity().onBackPressed();
                break;
            case R.id.clear_filter_button: {
                ((CameraActivity)getActivity()).clearFilter();
                break;
            }
            case R.id.filter_plus_button:
                ((CameraActivity)getActivity()).setFilterStrength(10);
                break;
            case R.id.filter_minus_button:
                ((CameraActivity)getActivity()).setFilterStrength(-10);
                break;
            case R.id.vignett_button:
                ((CameraActivity)getActivity()).setVignette();
                break;
            case R.id.blur_button:
                ((CameraActivity)getActivity()).setBlurVignette();
                break;
        }
    }

    @Override
    public void onFilterSelected(int position, ItemModel item) {
        ((CameraActivity)getActivity()).setFilter(item);
    }
}

File: sample-kotlin/app/src/main/java/com/seerslab/argear/sample/ui/FilterFragment.kt
package com.seerslab.argear.sample.ui

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.seerslab.argear.sample.R
import com.seerslab.argear.sample.api.ContentsResponse
import com.seerslab.argear.sample.databinding.FragmentFilterBinding
import com.seerslab.argear.sample.model.ItemModel
import com.seerslab.argear.sample.ui.adapter.FilterListAdapter
import com.seerslab.argear.sample.viewmodel.ContentsViewModel

class FilterFragment : Fragment(),
    View.OnClickListener,
    FilterListAdapter.Listener {

    companion object {
        private val TAG = FilterFragment::class.java.simpleName
    }

    private lateinit var filterListAdapter: FilterListAdapter
    private lateinit var contentsViewModel: ContentsViewModel
    private lateinit var rootView: FragmentFilterBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = DataBindingUtil.inflate(inflater, R.layout.fragment_filter, container, false)

        rootView.closeFilterButton.setOnClickListener(this)
        rootView.clearFilterButton.setOnClickListener(this)
        rootView.filterPlusButton.setOnClickListener(this)
        rootView.filterMinusButton.setOnClickListener(this)
        rootView.vignettButton.setOnClickListener(this)
        rootView.blurButton.setOnClickListener(this)

        val recyclerViewFilter: RecyclerView = rootView.filterRecyclerview
        recyclerViewFilter.setHasFixedSize(true)

        val filterLayoutManager = LinearLayoutManager(context)
        filterLayoutManager.orientation = LinearLayoutManager.HORIZONTAL
        recyclerViewFilter.layoutManager = filterLayoutManager

        filterListAdapter = FilterListAdapter(context, this)
        recyclerViewFilter.adapter = filterListAdapter

        return rootView.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        activity?.let { it ->
            contentsViewModel = ViewModelProvider(it).get(ContentsViewModel::class.java)
            contentsViewModel.contents.observe(
                viewLifecycleOwner,
                Observer<ContentsResponse?> { contentsResponse ->
                    contentsResponse?.categories?.let {
                        for (model in it) {
                            if (TextUtils.equals(model.title, "filters")) {
                                filterListAdapter.setData(model.items)
                                return@Observer
                            }
                        }
                    }
                })
        }
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.close_filter_button -> activity?.onBackPressed()
            R.id.clear_filter_button -> (activity as CameraActivity).clearFilter()
            R.id.filter_plus_button -> (activity as CameraActivity).setFilterStrength(10)
            R.id.filter_minus_button -> (activity as CameraActivity).setFilterStrength(-10)
            R.id.vignett_button -> (activity as CameraActivity).setVignette()
            R.id.blur_button -> (activity as CameraActivity).setBlurVignette()
        }
    }

    override fun onFilterSelected(position: Int, item: ItemModel?) {
        item?.let {
            (activity as CameraActivity).setFilter(item)
        }
    }
}